src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -31,8 +32,31 @@ internal static string ParseFileName(string path, ZipVersionMadeByPlatform madeB
             }
             else
             {
+                PathInternal.CheckInvalidPathChars(path);
                 return Path.GetFileName(path);
             }
         }
+
+        internal static readonly char[] InvalidPathChars =
```

Comment:
Rather than duplicating this data and logic, can this be separated out of PathInternal.Windows/Unix.cs into a PathInternal.CheckInvalidChars.Windows/Unix.cs, which could then be included here and anywhere else it's needed?